cvat/apps/engine/media_extractors.py

```diff
@@ -72,6 +72,52 @@ def save_image(self, k, dest_path):
         image.close()
         return width, height
 
+class PDFExtractor(MediaExtractor):
+    def __init__(self, source_path, dest_path, image_quality, step=1, start=0, stop=0):
+        if not source_path:
+            raise Exception('No PDF found')
+        from pdf2image import convert_from_path
+        self._temp_directory = tempfile.mkdtemp(prefix='cvat-')
+        super().__init__(
+            source_path=source_path[0],
+            dest_path=dest_path,
+            image_quality=image_quality,
+            step=1,
+            start=0,
+            stop=0,
+        )
+        self._dimensions = []
+        file_ = convert_from_path(self._source_path)
+        self._basename = os.path.splitext(os.path.basename(self._source_path))[0]
+        for page_num, page in enumerate(file_):
+            output = os.path.join(self._temp_directory, self._basename + f'{page_num}' + '.jpg')
+            self._dimensions.append(page.size)
+            page.save(output, 'JPEG')
+
+    def _get_imagepath(self, k):
+        img_path = os.path.join(self._temp_directory, self._basename + f'{k}' + '.jpg')
+        return img_path
+
+    def __iter__(self):
+        i = 0
+        while os.path.exists(self._get_imagepath(i)):
+            yield self._get_imagepath(i)
+            i += 1
+
+    def __del__(self):
+        if self._temp_directory:
+            shutil.rmtree(self._temp_directory)
+
+    def __getitem__(self, k):
+        return self._get_imagepath(k)
+
+    def __len__(self):
+        return len(os.listdir(self._temp_directory))
```

Comment:
let's calculate the length in the \_\_init\_\_ method.

Reply:
Fixed